xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -35,58 +35,35 @@
 import java.util.Map;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
-import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * Loads configuration information to bootstrap gRPC's integration of xDS protocol.
  */
-@ThreadSafe
 abstract class Bootstrapper {
 
   private static final String BOOTSTRAP_PATH_SYS_ENV_VAR = "GRPC_XDS_BOOTSTRAP";
 
-  static Bootstrapper newInsatnce() {
-    return new FileBasedBootstrapper();
+  private static final Bootstrapper DEFAULT_INSTANCE = new Bootstrapper() {
+    @Override
+    BootstrapInfo readBootstrap() throws IOException {
+      String filePath = System.getenv(BOOTSTRAP_PATH_SYS_ENV_VAR);
+      if (filePath == null) {
+        throw new IOException("Environment variable " + BOOTSTRAP_PATH_SYS_ENV_VAR + " not found.");
```

Comment:
"not found" -> "not defined"    (more accurate)

I know the design asks for using an environment variable but I was thinking whether in Java we can also use a Java system property when the env var is not defined. Makes it easy to specify the bootstrap file on the JVM command line and also unit test (since you cannot set an env var from code but you can set a system property AFAIK). Assuming @markdroth is okay with the suggestion?

Reply:
My gut reaction is that it would be better to be consistent across languages, so we should stick to an environment variable.  But I'd like input from @ejona86 on this.

Reply:
We will be consistent since env var will be supported. This is a "java-only" extension. One issue I see is if both are defined, how is that resolved...

Reply:
Changed to "not defined".

Changes for supporting Java system property should be done in a separate PR if designers agreed upon it.

Reply:
We want an environment variable, since we will be receiving this from the environment. It is _much_ easier to specify an environment variable when starting a container vs futzing with system properties.